OsmAnd-java/src/main/java/net/osmand/router/HHRoutePlanner.java

```diff
@@ -718,6 +718,18 @@ public boolean contains(LatLon p) throws IOException {
 			}
 			return false;
 		}
+
+		public boolean containsRegion(List<String> regionsWithAllRoutePoints) {
+			for (BinaryMapIndexReader reader : readers) {
```

Comment:
Needs to check section I believe not file names 
regionsContainsRoutePoints.get()